src/Microsoft.DotNet.Cli.Sln.Internal/SlnFile.cs

```diff
@@ -698,17 +698,34 @@ internal void ReadLine(string line, int currentLine)
 
         internal void Write(TextWriter writer)
         {
-            foreach (DictionaryEntry e in _values)
+            if (_isMetadata)
             {
-                if (!_isMetadata)
+                foreach (DictionaryEntry e in _values)
                 {
-                    writer.Write("\t\t");
+                    if (Id != null)
+                    {
+                        writer.Write(Id + ".");
+                    }
+                    writer.WriteLine(e.Key + " = " + e.Value);
```

Comment:
to be honest this is quite magic. I got no clue what's being written in here, could at minimum add some comment saying which part of sln this is producing?